### PR TITLE
New yaml config

### PR DESCRIPTION
**Pull request by rautesamtr**
This adds async_setup to the thermal_comfort integration and allows to
setup the platforms under integration domain key.
e.g.
```yaml
thermal_comfort:
  - sensor:
    - temperature_sensor: sensor.kitchen_temperature
      humidity_sensor: sensor.kitchen_humidity
```

With this commit the thermal_comfort integration supports both the old
legacy configuration under the platform key e.g. sensor and the new
configuration under integration domain key.

The `friendly_name` config key has been renamed to `name` and takes up
the role as both friendly name and identifier = entity_id.
This requires user intervention when converting the old
configurations to new ones.

e.g.
```yaml
sensor:
  - platform: thermal_comfort
    sensors:
      old_kitchen:
        friendly_name: Old Kitchen
        temperature_sensor: sensor.kitchen_temperature
        humidity_sensor: sensor.kitchen_humidity
```
would need to be changed to
```yaml
thermal_comfort:
  - sensor:
    - name: Old Kitchen
      temperature_sensor: sensor.kitchen_temperature
      humidity_sensor: sensor.kitchen_humidity
```

### CODE REVIEW

**Comment by rautesamtr** [comment]
Hey, @IATkachenko this PR would merges the separate SensorThermalComfort back together because with this changes it they wouldn't be needed anymore. Do you have anything specific in mind for which you'd need the separate class in the future?

**Comment by IATkachenko** [comment]
`_attr_device_info` was only one reason for splitting: we must have this for configuration via UI, but may not have in .yaml case.
`hass` was added as part of updating mechanism _(and I forgot to remove it)_, that was not used.

File: custom_components/thermal_comfort/sensor.py [hunk 1]
```diff
@@ -299,30 +311,38 @@ def id_generator(unique_id: str, sensor_type: str) -> str:
...
+        device_name: str = None,
+        icon_template: Template = None,
+        entity_picture_template: Template = None,
+        device_unique_id: str = None,
+    ) -> None:
         """Initialize the sensor."""
-        self._device = device
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, device_unique_id)},
+        )
```
Comment (IATkachenko) [inline custom_components/thermal_comfort/sensor.py:330]: Should `name` be dropped while merging classes back?

Comment (rautesamtr) [reply custom_components/thermal_comfort/sensor.py:330]: No my bad, thanks for pointing that out

**Comment by rautesamtr** [comment]
> `_attr_device_info` was only one reason for splitting: we must have this for configuration via UI, but may not have in .yaml case. `hass` was added as part of updating mechanism _(and I forgot to remove it)_, that was not used.

I think it is fine to set it also for yaml since it isn't read if the entity isn't loaded via config entry.

> Entity device info is only read if the entity is loaded via a config entry and the unique_id property is defined.

I moved `device_info`, `device_unique_id` and `device_name` to `DeviceThermalComfort` since they are not sensor specific.

**Comment by IATkachenko** [comment]
LGTM, from config_flow point of view and reading the code.

Will be able to apply changes for tests at own installation only in the evening, if it needed.

**Comment by rautesamtr** [comment]
Thanks for the quick review. I'll merge this for now. I'd appreciate it if you could test it a bit later.

**Comment by IATkachenko** [comment]
Configuration via UI works fine.

**Pull request merged**